src/Features/LanguageServer/Protocol/Handler/Diagnostics/DiagnosticSourceProviders/WorkspaceDiagnosticSourceHelpers.cs

```diff
@@ -5,22 +5,14 @@
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Linq;
-using System.Threading;
-using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Host;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.LanguageServer.Handler.Diagnostics;
 
-internal abstract class AbstractWorkspaceDiagnosticSourceProvider(string name) : IDiagnosticSourceProvider
+internal static class WorkspaceDiagnosticSourceHelpers
 {
-    public bool IsDocument => false;
-    public string Name => name;
-
-    public abstract ValueTask<ImmutableArray<IDiagnosticSource>> CreateDiagnosticSourcesAsync(RequestContext context, CancellationToken cancellationToken);
-
-    protected static IEnumerable<Project> GetProjectsInPriorityOrder(
-        Solution solution, ImmutableArray<string> supportedLanguages)
+    public static IEnumerable<Project> GetProjectsInPriorityOrder(this Solution solution, ImmutableArray<string> supportedLanguages)
```

Comment:
def do not have an extension like this on Solution.  This is a very diagnostics-specific functionality.  HAving it jsut be a static call into thsi helper class.